src/App.tsx

```diff
@@ -17,18 +46,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onSelect={setSelectedStatus}
+                onChange={setSelectedQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!isLoaded ? (
+                <Loader />
+              ) : (
+                <TodoList
+                  todos={todos}
+                  selectedTodo={selectedTodo}
+                  onSelect={setSelectedTodo}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal onClose={setSelectedTodo} todo={selectedTodo} />
```

Comment:
The naming convention for event handler props suggests using `onModalClose` instead of `onClose` to make it more descriptive and specific to the action.